### PR TITLE
Manage and edit stored filters

### PR DESCRIPTION
**Pull request by cproensa**
Stored filter edit feature
A lot of clean up in filter_api to allow for code reuse.
Some bug fixes that came in the way

The new pages may need some tuning, but the underlying changes make it easy to work from here on

### CODE REVIEW

File: lang/strings_english.txt [hunk 1]
```diff
@@ -1260,6 +1260,16 @@ $s_query_name_too_long = 'You cannot store a filter name with more than 64 chara
 $s_query_store_error = 'There was an error saving this filter.';
 $s_open_queries = 'Manage Filters';
 $s_query_delete_msg = 'Are you sure you wish to delete this filter?';
+$s_edit_filter = "Edit Filter";
```
Comment (atrol) [inline lang/strings_english.txt]: Single quotes

File: lang/strings_english.txt [hunk 2]
```diff
@@ -1260,6 +1260,16 @@ $s_query_name_too_long = 'You cannot store a filter name with more than 64 chara
...
+$s_owner = "Owner";
+$s_update_filter = 'Update filter';
```
Comment (atrol) [inline lang/strings_english.txt]: Single quotes

File: lang/strings_english.txt [hunk 3]
```diff
@@ -1260,6 +1260,16 @@ $s_query_name_too_long = 'You cannot store a filter name with more than 64 chara
 $s_query_store_error = 'There was an error saving this filter.';
 $s_open_queries = 'Manage Filters';
 $s_query_delete_msg = 'Are you sure you wish to delete this filter?';
+$s_edit_filter = "Edit Filter";
+$s_owner = "Owner";
```
Comment (atrol) [inline lang/strings_english.txt]: Single quotes

**Comment by vboctor** [comment]
@cproensa do you have pictures that show the results of these changes?

File: return_dynamic_filters.php [hunk 4]
```diff
@@ -139,8 +107,14 @@ function return_dynamic_filters_prepend_headers() {
 } else if( 'custom_field' == utf8_substr( $f_filter_target, 0, 12 ) ) {
 	# custom function
 	$t_custom_id = utf8_substr( $f_filter_target, 13, -7 );
-	return_dynamic_filters_prepend_headers();
-	print_filter_custom_field( $t_custom_id );
+	$t_cfdef = custom_field_get_definition( $t_custom_id, false );
```
Comment (atrol) [inline return_dynamic_filters.php]: ,false? custom_field_get_definition has just one parameter

File: view_all_set.php [hunk 5]
```diff
@@ -374,11 +94,6 @@
 	$f_per_page = config_get( 'default_limit_view' );
 }
 
-# combine sort settings
-#  (f_sort overrides f_sort_1 if set to keep old sorting code working in view_all_bug_inc)
-$f_sort = ( ( $f_sort_d != '' ) ? $f_sort_d : $f_sort_0 ) . ( ( $f_sort_1 != '' ) ? ',' . $f_sort_1 : '' );
-$f_dir = ( ( $f_dir_d != '' ) ? $f_dir_d : $f_dir_0 ) . ( ( $f_dir_1 != '' ) ? ',' . $f_dir_1 : '' );
-
```
Comment (atrol) [inline view_all_set.php]: $f_sort and $f_dir are accessed later in code
		$t_setting_arr[FILTER_PROPERTY_SORT_FIELD_NAME] = $f_sort;
		$t_setting_arr[FILTER_PROPERTY_SORT_DIRECTION] = $f_dir;

**Comment by atrol** [comment]
Unfortunately no commenting in GitHub for core/filter_api.php possible as you get:

> 2,051 additions, 1,321 deletions not shown because the diff is too large. Please use a local Git client to view these changes. 

Some obvious bugs in filter_api.php found by static code analysis:

Line 518: $t_view_type = $f_default_view_type;
$f_default_view_type is not defined at this place

Line 4479: echo '<option value="' . string_attribute( $t_item ) . '"';
$t_item is no longer defined after your other changes

Line 5436: $t_filter_input[$t_field_name] = gpc_get_int_array( $t_field_name, $t_filtert[$t_field_name] );
Should be $t_filter instead of $t_filtert

File: return_dynamic_filters.php (hunk 4 above)
Comment (cproensa) [inline return_dynamic_filters.php]: thanks, fixed to:
$t_cfdef = @custom_field_get_definition( $t_custom_id );

File: view_all_set.php (hunk 5 above)
Comment (cproensa) [inline view_all_set.php]: fixed by using filter_gpc_get() on all those situations
most of the view_all_set operatios, defined by "type" request parameter, can be generalized to do a filter_gpc_get

**Comment by cproensa** [comment]
> Some obvious bugs in filter_api.php found by static code analysis:

thanks, fixed those.

what tool are you using for analysis?

**Comment by atrol** [comment]
> what tool are you using for analysis?

PhpStorm

**Comment by cproensa** [comment]
> do you have pictures that show the results of these changes?

Manage filters, replacing the "query_view_page"

manage_filter_page

Clicking on the filter name, goes to edit filter.
This can lead to view_filters_page (only usage permission) , or to manage_filter_edit_page (edit permissions)

view_filters_page, shows a filter as a standalone page. The action allowed is to apply the filter as current.

view_filters_page

manage_filter_edit_page. can change filter properties, to be saved back into the stored filter id.
Name, public, and project linking can also be modified.

manage_filter_edit_page

**Comment by cproensa** [comment]
reminder: track changes from #0021611

**Comment by cproensa** [comment]
> reminder: track changes from #0021611

Confirmed, it was fixed in this PR

**Comment by cproensa** [comment]
closed in favour of updated PR #862

File: view_all_set.php [hunk 6]
```diff
@@ -374,11 +94,6 @@
 	$f_per_page = config_get( 'default_limit_view' );
```
Comment (dregad) [inline view_all_set.php]: While testing, I got a `SYSTEM NOTICE: 'Undefined variable: f_per_page' in 'view_all_set.php'` at line 93

**Pull request closed without merging**